collect_app/src/test/java/org/odk/collect/android/formlists/blankformlist/BlankFormListMenuDelegateTest.kt

```diff
@@ -0,0 +1,219 @@
+        menuDelegate.onOptionsItemSelected(RoboMenuItem(R.id.menu_refresh))
+
+        assertThat(ShadowToast.getLatestToast(), nullValue())
+    }
+
+    @Test
+    fun `clicking refresh for sync when device is offline shows error toast and does not sync`() {
+        whenever(networkStateProvider.isDeviceOnline).thenReturn(false)
+
+        val menuDelegate = createMenuDelegate()
+        menuDelegate.onOptionsItemSelected(RoboMenuItem(R.id.menu_refresh))
+
+        assertThat(
+            ShadowToast.getTextOfLatestToast(), `is`(activity.getString(R.string.no_connection))
+        )
+        verify(viewModel, never()).syncWithServer()
+    }
+
+    @Test
+    fun `clicking sort displays sorting dialog`() {
+        val menuDelegate = createMenuDelegate()
+        menuDelegate.onOptionsItemSelected(RoboMenuItem(R.id.menu_sort))
+
+        assertThat(ShadowDialog.getLatestDialog(), instanceOf(FormListSortingDialog::class.java))
+    }
+
+    @Test
+    fun `changing search text should set filterText in viewModel`() {
+        val menu = createdMenu()
+        val menuDelegate = createMenuDelegate()
+
+        menuDelegate.onCreateOptionsMenu(menuInflater, menu)
+        menuDelegate.onPrepareOptionsMenu(menu)
+
+        val searchView = (menu.findItem(R.id.menu_filter).actionView as SearchView).findViewById<SearchView.SearchAutoComplete>(androidx.appcompat.R.id.search_src_text)
+        searchView.setText("abc")
+
+        verify(viewModel).filterText = "abc"
+    }
+
+    @Test
+    fun `clicking search when matchExactly enabled hides refresh and sort and then collapsing search shows them again`() {
+        whenever(viewModel.isMatchExactlyEnabled()).thenReturn(true)
+
+        val menu = createdMenu()
+        val menuDelegate = createMenuDelegate()
+
+        menuDelegate.onCreateOptionsMenu(menuInflater, menu)
+        menuDelegate.onPrepareOptionsMenu(menu)
+
+        menu.findItem(R.id.menu_filter).expandActionView()
+
+        assertThat(menu.findItem(R.id.menu_refresh).isVisible, `is`(false))
+        assertThat(menu.findItem(R.id.menu_sort).isVisible, `is`(false))
+
+        menu.findItem(R.id.menu_filter).collapseActionView()
+
+        assertThat(menu.findItem(R.id.menu_refresh).isVisible, `is`(true))
+        assertThat(menu.findItem(R.id.menu_sort).isVisible, `is`(true))
+    }
+
+    @Test
+    fun `collapsing search when matchExactly not enabled does not show refresh`() {
+        whenever(viewModel.isMatchExactlyEnabled()).thenReturn(false)
+
+        val menu = createdMenu()
+        val menuDelegate = createMenuDelegate()
+
+        menuDelegate.onCreateOptionsMenu(menuInflater, menu)
+        menuDelegate.onPrepareOptionsMenu(menu)
+
+        menu.findItem(R.id.menu_filter).expandActionView()
+        menu.findItem(R.id.menu_filter).collapseActionView()
+
+        assertThat(menu.findItem(R.id.menu_refresh).isVisible, `is`(false))
+    }
+
+    @Test
+    fun `filterText should be cleared when menu is created`() {
```

Comment:
Are we not doing extra work to prevent the filter text from being maintained on rotation/configuration change though?

Reply:
I clear the value stored in the viewmodel https://github.com/getodk/collect/pull/5077/files#diff-80daf2f5f67255e3fec799d7214bb0507e04c6b5c4f87075b318b6417014cd99R65 but the UI part is cleared automatically because it's recreated.

Reply:
Right but instead, could we just not clear the value in the `ViewModel` and filter text would be retained on rotation for free?

Reply:
No because the search view is closed automatically and I don't want to reopen it like I don't want to reopen the sorting dialog (I think those things are not that important). So if I remove that clearing the result will be that we see closed search view but filtered forms.

Reply:
> So if I remove that clearing the result will be that we see closed search view but filtered forms.

Ah I see! I think adding a comment to the test explaining this would be helpful.

Reply:
Done.